historical_surveillance/views.py

```diff
@@ -0,0 +1,328 @@
+from django.contrib import messages
+from django.shortcuts import render, redirect
+from django.db.models import Count
+from .forms import AggregateEnrollmentForms, DistrictForms, SchoolForms, EnrollmentForms
+from .models import AggregateEnrollment, School, District, Enrollment
+import pandas as pd
+from .utils import get_plot, get_pairs
+
+
+# This is the view for the home page of this app
+def index(request):
+    return render(request, 'surv_home.html', {})
+
+
+# This function controls the district creation form view and process form data for save into the database
+def district(request):
+    error_message = None
+    form = DistrictForms()
+    data = District.objects.all()
+    if request.method == 'POST':
+        form = DistrictForms(request.POST)
+        if form.is_valid():
+            # change the created by column to automatically insert the logged in user
+            district_created_by = form.cleaned_data['created_by']
+            district_code = form.cleaned_data['District_Code']
+            district_name = form.cleaned_data['District_Name']
+            # change the updated by column to automatically insert the name of the logged in user updating the column
+            district_updated = form.cleaned_data['updated_by']
+            if not District.objects.filter(District_Code=district_code,
+                                         created_by=district_created_by,
+                                         updated_by=district_updated,
+                                         District_Name=district_name).exists():
+                form.save()
+                return redirect("/")
+
+            else:
+                error_message = "The record you are trying to create already exists."
+
+        stu = {
+
+            "error_message": error_message,
+
+        }
+        return render(request, 'district.html', stu)
+    return render(request, 'district.html', {'form': form, 'district_created': data})
+
+
+# This function controls the schools creation form view and process form data for save into the database
+def create_school(request):
```

Comment:
Just FYI - using django forms we can easily make. this view handle both create and update. 
https://docs.djangoproject.com/en/1.8/topics/forms/modelforms/

Making a django form like this is create:
`form = SchoolForms()`

But making a django form like this will edit
```
school = School.objects.get(id=id)
form = SchoolForms(instance=school)
```
All the other code is the same. 

The interesting thing is we can set it up where the school to edits' id comes from the URL.  So. we can. do a url like `/school/edit/1` and then edit school 1.

Reply:
I think it might be good to discuss as a team a pattern we should be using for URLs. Don't mind what pattern we pick just as long as we stay consistent.
Some possible variations
`/<resource>/1/edit`
`/<resource>/edit/1`
`/<resource>/1` <-- usually I've seen this one as the read only page of the resource

Reply:
Yes. We can note them them and do that in the after Next week.